### DISCUSSION TITLE
Node feature

### DISCUSSION THREAD

**Discussion by shrimonmuke0202** [pyg-team/pytorch_geometric#3420] [Q&A] [answered]
I want to calculate the average of node features for a graph. How can I do this?

**Comment by rusty1s** [comment, marked as answer]
What do you mean with average? If you want to find the average value for each feature across the node dimension, then you can simply run:
```python
data.x.mean(dim=0)
```

**Reply by shrimonmuke0202** [reply]
I want to use dgl.mean_nodes method in PyG.

**Reply by rusty1s** [reply]
I see. This is equivalent to [`global_mean_pool`](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.glob.global_mean_pool)
```python
x = global_mean_pool(data.x, data.batch)
```
in a mini-batch scenario.

**Reply by shrimonmuke0202** [reply]
I want h^{'} = \frac{1}{|v|}\sum_{i=1}^{g}(h_{i}) where h is the feature of each node and |v| is the number of nodes in a graph g.

**Reply by rusty1s** [reply]
Yes, that's exactly what `data.x.mean(dim=0)` (on a single graph) or `x = global_mean_pool(...)` in a mini-batch of graphs is doing.

**Comment by shrimonmuke0202** [comment]
Thanks for your reply.
